source/extensions/clusters/well_known_names.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+
+/**
+ * Well-known cluster types, this supersede the service discovery types
+ */
+class ClusterTypeValues {
+public:
+  // Refer to the :ref:`static discovery type<arch_overview_service_discovery_types_static>`
```

Comment:
nit: These comments don't get rendered, so I would remove all the ref linking stuff and just have normal comments.